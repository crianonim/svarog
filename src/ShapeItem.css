.shapes-list  > div:first-of-type .move-up-button {
    visibility: hidden;
    /* background-color: red; */

}

.shapes-list > div:last-of-type .move-down-button   {
    /* background-color: aquamarine; */
    visibility: hidden;
}

.selected-shape {
    background: wheat;
}

.edit-enable-btn {
    width:1.5rem;
}