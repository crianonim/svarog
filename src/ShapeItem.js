import React,{useState} from 'react';
import BasicAttrEditor from './BasicAttrEditor.js';
import './ShapeItem.css';

const ShapeItem = (props) => {
    const shape = props.shape;
    const [edited,setEdited] = useState(false);
    return (
     <div key={shape.id} className="flex-row shape-item " >
        <div className="move-up-down-buttons">
          <button onClick={()=>{
            props.duplicate(shape);
          }} className="dup-shape-button"
          title="Duplicate Shape"
          >
            *
          </button>
            <button title="Move Up"  className="move-up-button"
              onClick={() => {
                  props.shapeUp(shape);
              }}>&#x21E7;</button>
            <button title="Move Down" className="move-down-button" onClick={() => {
                props.shapeDown(shape);
              }}
            >&#x21E9;</button>
            {edited?
            <button title="Disable attribute editing" className="edit-enable-btn" onClick={()=>setEdited(false)}>--</button>:
            <button title="Enable attribute editing" className="edit-enable-btn" onClick={()=>setEdited(true)}>...</button>
          }
        </div>
        <BasicAttrEditor
          element={shape.shape}
          attrs={shape.attributes}
          edited={edited}
          changed={attrs => {
            shape.attributes=attrs;
            props.changed(shape);
          }}
        />
       
    </div>
    );
}

export default ShapeItem;
