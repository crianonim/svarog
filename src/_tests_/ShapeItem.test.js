import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ShapeItem from "../components/ShapeItem";



it("duplicate button can be clicked",()=>{
    const duplicate=jest.fn();
    const wrapper=shallow(<ShapeItem duplicate={duplicate} shape={{shape:"circle",attributes:{"x":10,"y":20,"r":30}}}/>) 
    wrapper.find(".move-up-down-buttons button").first().simulate('click');
    expect(duplicate).toHaveBeenCalledTimes(1);
})