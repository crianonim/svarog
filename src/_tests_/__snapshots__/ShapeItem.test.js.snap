// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div
  className="flex-row shape-item "
>
  <div
    className="move-up-down-buttons"
  >
    <button
      className="button is-small is-primary"
      onClick={[Function]}
      title="Duplicate Shape"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f0c5",
              "M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z",
            ],
            "iconName": "copy",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <button
      className="move-up-button button is-small is-info"
      onClick={[Function]}
      title="Move Up"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f062",
              "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
            ],
            "iconName": "arrow-up",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <button
      className="move-down-button  button is-small is-info"
      onClick={[Function]}
      title="Move Down"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f063",
              "M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z",
            ],
            "iconName": "arrow-down",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
  </div>
  <span
    onClick={[Function]}
  >
    <AttrList
      attrs={
        Object {
          "r": 30,
          "x": 10,
          "y": 20,
        }
      }
      element="circle"
    />
  </span>
</div>
`;
